examples/src/llms/googlegenerativeai.ts

```diff
@@ -0,0 +1,67 @@
+import fs from "fs";
+import { HarmCategory, HarmBlockThreshold } from "@google/generative-ai";
+import { GoogleGenerativeAI } from "langchain/llms/googlegenerativeai";
+import { HumanMessage } from "@langchain/core/messages";
+
+/*
+ * Before running this, you should make sure you have created a
+ * Google Cloud Project that has `generativelanguage` API enabled.
+ *
+ * You will also need to generate an API key and set
+ * an environment variable GOOGLE_API_KEY
+ *
+ */
+
+// Text
+const model = new GoogleGenerativeAI({
+  modelName: "gemini-pro",
+  maxOutputTokens: 2048,
+  safetySettings: [
+    {
+      category: HarmCategory.HARM_CATEGORY_HARASSMENT,
+      threshold: HarmBlockThreshold.BLOCK_LOW_AND_ABOVE,
+    },
+  ],
+});
+const res = await model.call(
+  "What would be a good company name for a company that makes colorful socks?"
+);
+console.log(res);
+
+// Multi-modal
+const vision = new GoogleGenerativeAI({
+  modelName: "gemini-pro-vision",
+  maxOutputTokens: 2048,
+  safetySettings: [
+    {
+      category: HarmCategory.HARM_CATEGORY_HARASSMENT,
+      threshold: HarmBlockThreshold.BLOCK_LOW_AND_ABOVE,
+    },
+  ],
+});
+const image = fs.readFileSync("./image.png").toString("base64");
+const image2 = fs.readFileSync("./image2.png").toString("base64");
+const input2 = [
```

Comment:
Should this be a chat model then?

Reply:
From the API perspective it's the same handler which serves single-turn and multi-turn conversations.

It's more of a question should LLM support multi-modal inputs?

I just wanted to be consistent with Python counterpart: https://github.com/langchain-ai/langchain/pull/14557

Reply:
@jacoblee93 Refactoring this as a chat model.